HACKING.md

```diff
@@ -108,6 +108,30 @@ Language features:
 
 There is a lot of code in the project that doesn't follow these conventions. Please feel free to fix it, preferably not mixing up style and logical changes in the same PR.
 
+Testing
+-------
+We strive for a good test coverage of the project, and while we're not there yet, the current policy is to add tests for all the bugs we fix, as long as the fix is testable. E.g. graphical glitches are generally very hard to test, but we have the infractructure to test game logic and small isolated classes.
+
+Tests in OpenEnroth fall into two categories:
+* Unit tests. These are a standard breed of tests, written using Google Test. You can see some examples in `src/Utility/Tests`.
+* Game tests. If you're familiar with how testing is done these days for complex mobile apps, then you can consider game tests a variation of UI tests that's specific to our project. Game tests need game assets to run, and are currently **not** run in github actions.
+
+Game tests work by instrumenting the engine, and then running test code in between the game frames. This code usually sends events to the engine (e.g. mouse clicks), which are then processed by the engine in the next frame, but it can do pretty much anything else – all of engine's data is accessible and writable from inside the game test.
+
+As typing out all the events to send inside the test code can be pretty tedious, we also have an event recording functionality, so that the test creation workflow usually looks as follows:
+* Load a save that reproduces the bug that you've just fixed.
+* Press `Ctrl+Shift+R` to start recording an event trace. Check logs to make sure that trace recording has started.
+* Perform the steps that used to reproduce the bug.
+* Press `Ctrl+Shift+R` again to stop trace recording. You will get two files generated in the current folder – `trace.json` and `trace.mm7`.
+* Copy these files to the [OpenEnroth_TestData](https://github.com/OpenEnroth/OpenEnroth_TestData) repo and update the reference tag in the corresponding [CMakeLists.txt](https://github.com/OpenEnroth/OpenEnroth/blob/master/test/Bin/GameTest/CMakeLists.txt#L21).
```

Comment:
Who has access to TestData? Will non members need to do a PR instead?

Reply:
Yeah you're right, non-members will need to do a PR